src/query.c

```diff
@@ -446,7 +467,79 @@ extractClauseFromNullTest(Relids base_relids,
 	}
 }
 
+/*
+ *	Convert a "BoolExpr" (IS TRUE, IS FALSE, IS NOT TRUE, IS NOT FALSE)
+ *	to the corresponding qualifier.
+ */
+void extractClauseFromBooleanTest(Relids base_relids, BooleanTest *node, List **quals) 
+{
+	elog(DEBUG3, "entering extractClauseFromBooleanTest()");
+	if (IsA(node->arg, Var))
+	{
+		Var		   *var = (Var *) node->arg;
+		MulticornBaseQual *result;
+		char	   *opname = NULL;
+		Expr	   *val;
 
+		if (var->varattno < 1)
+		{
+			return;
+		}
+		switch (node->booltesttype)
+		{
+			case IS_TRUE:
+				opname = "IS";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			case IS_FALSE:
+				opname = "IS";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(false), false, true);
+				break;
+			case IS_NOT_TRUE:
+				opname = "IS NOT";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			case IS_NOT_FALSE:
+				opname = "IS NOT";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			default:
+				/* IS UNKNOWN, IS NOT UNKNOWN */
+				elog(ERROR, "unsupported boolean test type %d", node->booltesttype);
+		}
+		result = makeQual(var->varattno, opname,
+						  (Expr *) val, false, false);
+		*quals = lappend(*quals, result);
+	}
+}
+
+void extractClauseFromVar(
+#if PG_VERSION_NUM >= 140000
+    PlannerInfo *root,
+#endif
+	Relids base_relids, Var *var, List **quals)
+{
+	MulticornBaseQual *result;
+	Expr *true_expr;
+    if (!bms_is_subset(pull_varnos(
+#if PG_VERSION_NUM >= 140000
+		root,
+#endif
+		(Node *) var), base_relids)) {
+        return;
+	}
+
+	true_expr = (Expr *) makeConst(BOOLOID,  // Type OID for boolean
+		-1,       // typmod
+		InvalidOid, // collation
+		sizeof(bool), // constlen
+		BoolGetDatum(true), // the actual value
+		false,     // isnull
+		true);     // constbyval
+
+	result = makeQual(var->varattno, "=", true_expr, false, false);
```

Comment:
What is the reasoning for using `"="` as an operation name here, rather than `IS`?  As an implementor on the python side you'd have to support a couple different operators even though they seem logically the same.

Reply:
Yup, that makes sense. 👍